eureka/niriss_class.py

```diff
@@ -0,0 +1,516 @@
+        ----------
+        filename : str
+           The name of the F277W FITS file.
+
+        Attributes
+        ----------
+        f277 : np.ndarray
+           Science images from this filter.
+        """
+        hdu = fits.open(os.path.join(self.data_dir,
+                                     filename) )
+        self.f277 = hdu[1].data + 0.0
+
+        hdu.close()
+        return
+
+
+    def clean_up(self):
+        """
+        The `clean_up` routine interpolated over bad pixel values,
+        which are marked in the data quality images (`self.dq`).
+        This routine removes bad quality pixels from the following
+        images:
+           - `self.data`
+           - `self.err`
+           - `self.var`
+        """
+        print('Cleaning data . . .')
+        self.data = interpolating_image(self.data, mask=self.dq)
+        self.median = np.nanmedian(self.data, axis=0)
+        print('Cleaning error . . .')
+        self.err  = interpolating_image(self.err,  mask=self.dq)
+        print('Cleaning variance . . .')
+        self.var  = interpolating_image(self.var,  mask=self.dq)
+        self.median = interpolating_image(self.median, mask=np.nanmedian(self.dq,axis=0))
+
+
+    def map_wavelength(self, orders=[1,2,3]):
+        """
+        Retrieves the 2D wavelength maps for the first and
+        second orders.
+
+        Parameters
+        ----------
+        orders : np.array, list, optional
+           A list of which orders to retrieve the wavelength
+           solutions for. Default is [1,2,3]. Options are any
+           single order or combinations of these orders.
+
+        Attributes
+        ----------
+        wavelength_map : np.ndarray
+        """
+        wmap = wavelength(os.path.join(self.data_dir, self.filename),
+                          orders, inclass=True)
+        self.wavelength_map = wmap + 0.0
+
+
+    def map_trace(self, method='profile', ref_filename=None, isplots=0):
+        """
+        Calculates the trace of the first and second NIRISS
+        orders.
+
+        Parameters
+        ----------
+        method : str, optional
+           Decision on which trace extraction routine to run.
+           Options are: `edges` (uses a canny-edge detection
+           routine), `centers` (uses the spatial profile), and
+           `ref` (uses the STScI JWST reference frame).
+        ref_filename : str, optional
+           The name of the reference frame containing the order
+           trace x,y position values. Default is None. This is
+           a required parameter if you are running `method=='ref'`.
+
+        Attributes
+        ----------
+        trace_ear : astropy.table.Table
+           Astropy table with x,y coordinates for each order.
+           `trace_ear` is initialized when using the method `edges`.
+        trace_edge : astropy.table.Table
+           Astropy table with x,y coordinates for each order.
+           `trace_edge` is initialized when using the method `centers`.
```

Comment:
If these are newly added attributes, they should be explained up at the top of the class, not here

Reply:
added